code/modules/mob/living/carbon/human/species_types/psyphoza.dm

```diff
@@ -0,0 +1,359 @@
+//keep this type-
+/atom/movable/screen/fullscreen/blind/psychic
+	icon_state = "trip"
+	icon = 'icons/mob/psychic.dmi'
+	//The color we return to after going black & back.
+	var/origin_color = "#111"
+
+/atom/movable/screen/fullscreen/blind/psychic/Initialize(mapload)
+	. = ..()
+	filters += filter(type = "radial_blur", size = 0.012)
+	color = origin_color
+
+//And this type as a seperate type-path to avoid issues with animations & locate()
+/atom/movable/screen/fullscreen/blind/psychic_highlight
+	icon_state = "trip"
+	icon = 'icons/mob/psychic.dmi'
+	plane = PSYCHIC_PLANE
+	blend_mode = BLEND_INSET_OVERLAY
+	///Index for visual setting
+	var/visual_index = 0
+	
+/atom/movable/screen/fullscreen/blind/psychic_highlight/Initialize(mapload)
+	. = ..()
+	filters += filter(type = "bloom", size = 2, threshold = rgb(85,85,85))
+	filters += filter(type = "radial_blur", size = 0.012)
+	cycle_visuals()
+
+/atom/movable/screen/fullscreen/blind/psychic_highlight/proc/cycle_visuals()
+	++visual_index
+	//Reset animations
+	animate(src, color = "#fff")
+	//Set animation
+	switch(visual_index)
+		if(1) //Rainbow
+			color = "#f00" // start at red
+			animate(src, color = "#ff0", time = 0.3 SECONDS, loop = -1)
+			animate(color = "#0f0", time = 0.3 SECONDS)
+			animate(color = "#0ff", time = 0.3 SECONDS)
+			animate(color = "#00f", time = 0.3 SECONDS)
+			animate(color = "#f0f", time = 0.3 SECONDS)
+			animate(color = "#f00", time = 0.3 SECONDS)
+		if(2) //Custom
+			color = input(usr,"","Choose Color","#fff") as color|null
+	//Wrap index back around
+	visual_index = visual_index >= 2 ? 0 :  visual_index
+
+/datum/action/change_psychic_visual
+	name = "Change Psychic Sense"
+	desc = "Change the visual style of your psychic sense."
+	icon_icon = 'icons/mob/actions.dmi'
+	button_icon_state = "unknown"
+	///Ref to the overlay - hard del edition
+	var/atom/movable/screen/fullscreen/blind/psychic_highlight/psychic_overlay
+
+/datum/action/change_psychic_visual/Trigger()
+	. = ..()
+	if(!psychic_overlay)
+		psychic_overlay = locate(/atom/movable/screen/fullscreen/blind/psychic_highlight) in owner?.client?.screen
+	psychic_overlay?.cycle_visuals()
+
+/obj/effect/psyphoza_spores
+	name = "psyphoza sprore deposit"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shrooms"
+	invisibility = SHROOM_DEPOSIT_INVISIBILITY
+	///Internal resources for healing
+	var/resources = 0
+	///Internal resource limit
+	var/resource_limit = 20
+	///Amount of upgrades
+	var/upgrades = 1
+	///Limit to upgrades
+	var/upgrade_limit = 3
+	///The calling card for this particular deposit
+	var/message = ""
+
+/obj/effect/psyphoza_spores/Initialize(mapload, new_message)
+	. = ..()
+	message = new_message
+	START_PROCESSING(SSobj, src)
+
+/obj/effect/psyphoza_spores/attack_hand(mob/living/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Available resources: [resources] \nThe deposit echos: '[message]'</span>")
+
+/obj/effect/psyphoza_spores/process(delta_time)
+	if(world.time % 3 == 0)
```

Comment:
You should work around delta_time here.
You do not have a guarantee that this will be fired every x time, so you might miss some ticks when world.time would be divisible by 3.